api/v1alpha1/apidefinition_types.go

```diff
@@ -163,8 +168,32 @@ type URLRewriteMeta struct {
 	// MatchPattern is a regular expression pattern to match the path
 	MatchPattern string `json:"match_pattern"`
 	// RewriteTo is the target path on the upstream, or target URL we wish to rewrite to
-	RewriteTo string           `json:"rewrite_to"`
-	Triggers  []RoutingTrigger `json:"triggers"`
+	// +optionall
+	RewriteTo string `json:"rewrite_to,omitempty"`
+	// RewriteToLoop serves as rewrite_to but used when rewriting to looping
+	// targets. When rewrite_to and rewrite_to_loop are both provided then
+	// rewrite_to will take rewrite_to_loop
+	// +optionall
+	RewriteToLoop *LoopTarget `json:"rewrite_to_loop,omitempty"`
```

Comment:
The API Definition specifies `internal: true`.
So rather than calling it `RewriteToLoop`, we should call it `RewriteInternalTarget` or similar?
Further - how about `namespace/name`?

I want to rewrite to a target in a different k8s namespace? Do we want to enable this capability?

Reply:
done 33def54